### PR TITLE
Add saleae capture and script to parse

### PR DESCRIPTION
**Pull request by jonmagic**
``` json
$ script/saleae saleae_captures/inserting-cd.csv
[
  {
    "bits": "1100001101001000000100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1000000100",
    "milliseconds_to_next_message": 101.56066666666707
  },
  {
    "bits": "111011110100110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 14.841000000000104
  },
  {
    "bits": "111001110100110011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 7.512500000000033
  },
  {
    "bits": "1100001101001000000100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1000000100",
    "milliseconds_to_next_message": 4.040499999999891
  },
  {
    "bits": "111001101110000100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.070583333333879
  },
  {
    "bits": "110000110100000011",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "000011",
    "milliseconds_to_next_message": 420.45775000000106
  },
  {
    "bits": "1100001011111110010101",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 795.9443333333343
  },
  {
    "bits": "110000110100100010",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "100010",
    "milliseconds_to_next_message": 34.89825000000035
  },
  {
    "bits": "11100111010001110010",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 9.06616666666693
  },
  {
    "bits": "1100001101000000100100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "0000100100",
    "milliseconds_to_next_message": 2615.1209999999996
  },
  {
    "bits": "110000110100010111",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "010111",
    "milliseconds_to_next_message": 4.367833333334126
  },
  {
    "bits": "1100001101001101100011",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "1101100011",
    "milliseconds_to_next_message": 4.3644166666672035
  },
  {
    "bits": "110000110100100010000101",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "100010000101",
    "milliseconds_to_next_message": 408.0700833333335
  },
  {
    "bits": "11000011010000001010",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "00001010",
    "milliseconds_to_next_message": 4.920499999999883
  },
  {
    "bits": "101000110100011011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.362833333332539
  },
  {
    "bits": "101000110100000101",
    "type": "minutes",
    "prefix": "10100011010000010",
    "bits_after_prefix": "1",
    "milliseconds_to_next_message": 4.363833333333567
  },
  {
    "bits": "101000110100100100",
    "type": "seconds",
    "prefix": "10100011010010010",
    "bits_after_prefix": "0",
    "milliseconds_to_next_message": 35.88158333333347
  },
  {
    "bits": "11100111010000111100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.0709166666663705
  },
  {
    "bits": "101000110100011011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.365166666666199
  },
  {
    "bits": "101000110100000101",
    "type": "minutes",
    "prefix": "10100011010000010",
    "bits_after_prefix": "1",
    "milliseconds_to_next_message": 4.366833333333098
  },
  {
    "bits": "101000110100100100",
    "type": "seconds",
    "prefix": "10100011010010010",
    "bits_after_prefix": "0",
    "milliseconds_to_next_message": 25.938499999999642
  },
  {
    "bits": "11100111010001101100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 226.8849166666671
  },
  {
    "bits": "1100001101000000100100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "0000100100",
    "milliseconds_to_next_message": 218.77549999999957
  },
  {
    "bits": "11000011010000001100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "00001100",
    "milliseconds_to_next_message": 4.925000000000068
  },
  {
    "bits": "101000110100011011",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 4.36324999999993
  },
  {
    "bits": "101000110100000101",
    "type": "minutes",
    "prefix": "10100011010000010",
    "bits_after_prefix": "1",
    "milliseconds_to_next_message": 4.366166666666338
  },
  {
    "bits": "101000110100100100",
    "type": "seconds",
    "prefix": "10100011010010010",
    "bits_after_prefix": "0",
    "milliseconds_to_next_message": 17.9669166666665
  },
  {
    "bits": "111001101110000100",
    "type": null,
    "prefix": null,
    "bits_after_prefix": null,
    "milliseconds_to_next_message": 6.0749166666660415
  },
  {
    "bits": "11000011010000001100",
    "type": "from cd-changer",
    "prefix": "110000110100",
    "bits_after_prefix": "00001100",
    "milliseconds_to_next_message": 893.3025000000016
  },
  {
    "bits": "10100011010010010101",
    "type": "seconds",
    "prefix": "10100011010010010",
    "bits_after_prefix": "101",
    "milliseconds_to_next_message": 939.8252500000001
  },
  {
    "bits": "10100011010010010011",
    "type": "seconds",
    "prefix": "10100011010010010",
    "bits_after_prefix": "011",
    "milliseconds_to_next_message": 0
  }
]
```

``` bash
$ script/saleae saleae_captures/inserting-cd.csv | jq '.[] | .bits'
"1100001101001000000100"
"111011110100110010"
"111001110100110011"
"1100001101001000000100"
"111001101110000100"
"110000110100000011"
"1100001011111110010101"
"110000110100100010"
"11100111010001110010"
"1100001101000000100100"
"110000110100010111"
"1100001101001101100011"
"110000110100100010000101"
"11000011010000001010"
"101000110100011011"
"101000110100000101"
"101000110100100100"
"11100111010000111100"
"101000110100011011"
"101000110100000101"
"101000110100100100"
"11100111010001101100"
"1100001101000000100100"
"11000011010000001100"
"101000110100011011"
"101000110100000101"
"101000110100100100"
"111001101110000100"
"11000011010000001100"
"10100011010010010101"
"10100011010010010011"
```

/cc @jch

### CODE REVIEW

File: script/saleae [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+# Usage: script/saleae /path/to/captured_data.csv
+#        script/saleae data.csv | jq '.[] | .message'
```
Comment (jch) [inline script/saleae:3]: :+1: on including jq usage

File: script/saleae [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+messages = []
+message = ""
+
+csv.each_with_index do |row, index|
+  if signal(row) == 0
+    # check width of pulse and record 0 or 1
+    width_of_pulse = seconds_to_micros(timestamp(csv[index+1]) - timestamp(row))
+
+    case width_of_pulse
+    when 70..110
```
Comment (jch) [inline script/saleae:61]: nice use of ranges

File: script/saleae (hunk 1 above)
Comment (jonmagic) [inline script/saleae:3]: Just realized I forgot to update the `.message` to be `.bits`... gonna commit directly to master :grin:

**Comment by jch** [comment]
Good stuff!

**Pull request merged**